### PR TITLE
[processing] vector outputs can be saved in any GDAL-supported format

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Manually apllied changes by @medspx from #5726. Thanks to Médéric RIBREUX for this feature.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/plugins/processing/algs/grass7/Grass7Algorithm.py [hunk 1]
```diff
@@ -788,16 +808,19 @@ def loadVectorLayer(self, name, layer, external=False):
...
-        Creates a dedicated command to export a raster from
-        temporary GRASS DB into a file via gdal.
-        :param grassName: name of the parameter
-        :param fileName: file path of raster layer
-        :param colorTable: preserve color Table.
+        Creates a dedicated command to export a vector from
+        a QgsProcessingParameter.
+        :param name: name of the parameter.
+        :param context: parameters context.
+        :param layer: for vector with multiples layers,exports only one layer.
```
Comment (DelazJ) [inline python/plugins/processing/algs/grass7/Grass7Algorithm.py]: s/multiples/multiple
missing spacing after the comma
For consistency with other descriptions (see eg L845), should it be "exports" or "export"?

**Comment by alexbruy** [comment]
@medspx can you take a look at this?

**Comment by nyalldawson** [comment]
Nice! (I'm not familiar enough with grass to know if this is safe though, so don't consider this a code review ;))

**Comment by alexbruy** [comment]
Neither am I. Let's wait a bit, maybe someone with better GRASS knowledge can look at this.

**Pull request closed without merging**